src/utils/common-expression-helpers.ts

```diff
@@ -260,17 +260,14 @@ export class CommonExpressionHelpers {
   };
 
   calcGravida = (parityTerm, parityAbortion) => {
-    let gravida = 0;
-    if (parityTerm === parseInt(parityTerm)) {
-      gravida = parityTerm + 1;
-    }
-    if (parityAbortion === parseInt(parityAbortion)) {
-      gravida = parityAbortion + 1;
-    }
-    if (parityAbortion === parseInt(parityAbortion) && parityTerm === parseInt(parityTerm)) {
-      parityAbortion + parityTerm + 1;
+    const term = parseInt(parityTerm, 10);
+    const abortion = parseInt(parityAbortion, 10);
+
+    if (isNaN(term) || isNaN(abortion)) {
```

Comment:
```suggestion
    if (!Number.isInteger(term) || !Number.isInteger(abortion)) 
```

Perhaps [Number.isInteger](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) is a better check here since we're specifically dealing with whole numbers. Unlike `isNaN`, which allows any valid number (including decimals), `Number.isInteger` will reject decimals. Additionally, `isNaN` attempts to coerce the input into a number before checking, which can lead to unintended behavior, while `Number.isInteger` will return false for any non-number types. If we decide to consider decimal inputs in the future, reverting to `isNaN` would be more appropriate.

Reply:
Thanks @denniskigen, this has been resolved.

Reply:
Per our conversation from Coffee Break call, you might need to consider the current pregnancy in your calculation @arodidev.